### ISSUE TITLE
Client.query(config, values, callback) sets config.callback, causing subsequent invocations to fail

### ISSUE THREAD

**Issue by christerswahn** [brianc/node-postgres#2651]
Client.query(), when using promises (no callback), will via normalizeQueryConfig() set config.callback, which causes subsequent invocations of Client.query() with the same config to fail silently - no error message, just returning undefined.

Passed config should never be modified, this is a client-owned object.

Functions such as this should never fail silently - either return a valid value / defined error value, or raise exception.

**Comment by brianc** [comment]
> Passed config should never be modified, this is a client-owned object.

I agree with you.  Do you have code to reproduce this?  I can fix p quick.

**Comment by jbilous** [comment]
Wanted to bump this, I'm seeing very similar behavior. I was switching from callbacks to promises and noticed client.query() began returning undefined periodically instead of returning a promise.

**Comment by Lukas-Batema** [comment]
> Wanted to bump this, I'm seeing very similar behavior. I was switching from callbacks to promises and noticed client.query() began returning undefined periodically instead of returning a promise.

Same here, I would love to see this fixed, as the issue heavily breaks my projects using this.
